charmcraft/application/main.py

```diff
@@ -99,7 +100,7 @@ def _extra_yaml_transform(
             for field in const.METADATA_YAML_MIGRATE_FIELDS:
                 yaml_data.setdefault(field, metadata_yaml.get(field))
 
-        return yaml_data
+        return apply_extensions(self._work_dir, yaml_data)
```

Comment:
```suggestion
        return yaml_data
```

Merge conflict here :-)

The extensions get applied on line 76 so this is no longer needed.